include/zephyr/pm/policy.h

```diff
@@ -158,6 +165,43 @@ void pm_policy_latency_changed_subscribe(struct pm_policy_latency_subscription *
  */
 void pm_policy_latency_changed_unsubscribe(struct pm_policy_latency_subscription *req);
 
+/**
+ * @brief Register an event.
+ *
+ * Events in the power-management policy context are defined as any source that
+ * will wake up the system at a known time in the future. By registering such
+ * event, the policy manager will be able to decide whether certain power states
+ * are worth entering or not.
+ *
+ * @note Expired events need to be unregistered manually using
+ * pm_policy_event_unregister().
+ *
+ * @param evt Event.
+ * @param time When the event will occur, in microseconds from now.
+ *
+ * @see pm_policy_event_unregister
+ */
+void pm_policy_event_register(struct pm_policy_event *evt, uint32_t time);
```

Comment:
I think it is worth to change the variable to `time_us` just to make ti clear the resolution.

Reply:
I think that makes sense, however, I'd also adjust other calls like the latency ones (which use `time` for the same)